src/java.base/share/classes/jdk/internal/util/SystemProps.java

```diff
@@ -88,9 +88,12 @@ public static Map<String, String> initProperties() {
             put(props, "file.encoding", nativeEncoding);
         }
 
-        // "stdout/err.encoding", prepared for System.out/err. For compatibility
-        // purposes, substitute them with "sun.*" if they don't exist. If "sun.*" aren't
-        // available either, fall back to "native.encoding".
+        // Encoding properties for stdin, stdout, and stderr. For stdout and stderr,
+        // check "sun.*.encoding" properties before falling back to the
```

Comment:
Could explain about sun* a bit more (just for compatibility) and stdin.encoding is free from it.

Reply:
I think the comment could be clearer. As I understand it, it's just trying to saying that the legacy sun.stdout.encoding/sun.stderr.encoding properties for stdout/stderr only, no equivalent for stdin.

Reply:
Yeah the comment was a bit terse so I tried to clarify it.

Reply:
Thanks, it looks okay now.